sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/GenericTypeParameterUnused.cs

```diff
@@ -52,6 +52,7 @@ protected override void Initialize(SonarAnalysisContext context)
                     {
                         var methodDeclaration = c.Node as MethodDeclarationSyntax;
```

Comment:
I find this code quite confusing, I think it's a good time to refactor.
Lines 53-72 have a lot of local variables which are not used after line 73: `methodDeclaration`, `classDeclaration` and now `localFunctionDeclaration`. They're only useful to get the `helper` which is later used.
So maybe it's a good hint to extract a method to get the `helper` based on the `c.Node`

Reply:
Have you checked all commits? This was handled already in 2a2c938441beb24aa54be4973f4c9a246e1d1073.